libs/pika/threading_base/src/scheduler_mode.cpp

```diff
@@ -0,0 +1,43 @@
+//  Copyright (c) 2022  ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <pika/config.hpp>
+#include <pika/threading_base/thread_pool_base.hpp>
+
+#include <cstdint>
+
+namespace pika::threads {
+
+    scheduler_mode operator&(scheduler_mode sched1, scheduler_mode sched2)
+    {
+        return static_cast<scheduler_mode>(static_cast<std::uint32_t>(sched1) &
+            static_cast<std::uint32_t>(sched2));
+    }
+
+    scheduler_mode operator|(scheduler_mode sched1, scheduler_mode sched2)
+    {
+        return static_cast<scheduler_mode>(static_cast<std::uint32_t>(sched1) |
+            static_cast<std::uint32_t>(sched2));
+    }
+
+    bool operator!=(scheduler_mode sched, int i)
+    {
+        return static_cast<std::uint32_t>(sched) !=
+            static_cast<std::uint32_t>(i);
+    }
+
+    bool operator==(scheduler_mode sched, int i)
+    {
+        return static_cast<std::uint32_t>(sched) ==
+            static_cast<std::uint32_t>(i);
+    }
```

Comment:
Do we still need these?

Reply:
Where do we need them? More importantly, are they needed by users of `scheduler_mode` (my guess is no)?

Reply:
Thanks. Yeah, in that case please remove the `!=` and `==` operators and I think you should be able to compare to `scheduler_mode{}` instead.

Reply:
Enums are already comparable, so you don't need any overloads for that.